applications/PfemFluidDynamicsApplication/custom_constitutive/fluid_laws/temperature_dependent/frictional_viscoplastic_temperature_dependent_2D_law.cpp

```diff
@@ -0,0 +1,167 @@
+
+    ConstitutiveLaw::Pointer FrictionalViscoplasticTemperatureDependent2DLaw::Clone() const { return Kratos::make_shared<FrictionalViscoplasticTemperatureDependent2DLaw>(*this); }
+
+    //*********************************DESTRUCTOR*********************************
+    //****************************************************************************
+
+    FrictionalViscoplasticTemperatureDependent2DLaw::~FrictionalViscoplasticTemperatureDependent2DLaw() {}
+
+    std::string FrictionalViscoplasticTemperatureDependent2DLaw::Info() const { return "FrictionalViscoplasticTemperatureDependent2DLaw"; }
+
+    //******************CHECK CONSISTENCY IN THE CONSTITUTIVE LAW******************
+    //*****************************************************************************
+
+    int FrictionalViscoplasticTemperatureDependent2DLaw::Check(const Properties &rMaterialProperties, const GeometryType &rElementGeometry,
+                                                               const ProcessInfo &rCurrentProcessInfo) const
+    {
+
+        if (rMaterialProperties[DYNAMIC_VISCOSITY] < 0.0)
+        {
+            KRATOS_ERROR << "Incorrect or missing DYNAMIC_VISCOSITY provided in process info for FrictionalViscoplasticTemperatureDependent2DLaw: "
+                         << rMaterialProperties[DYNAMIC_VISCOSITY] << std::endl;
+        }
+
+        if (rMaterialProperties[INTERNAL_FRICTION_ANGLE] < 0.0)
+        {
+            KRATOS_ERROR << "Incorrect or missing INTERNAL_FRICTION_ANGLE provided in process info for FrictionalViscoplasticTemperatureDependent2DLaw: "
+                         << rMaterialProperties[INTERNAL_FRICTION_ANGLE] << std::endl;
+        }
+
+        if (rMaterialProperties[COHESION] < 0.0)
+        {
+            KRATOS_ERROR << "Incorrect or missing COHESION provided in process info for FrictionalViscoplasticTemperatureDependent2DLaw: "
+                         << rMaterialProperties[COHESION] << std::endl;
+        }
+
+        if (rMaterialProperties[ADAPTIVE_EXPONENT] < 0.0)
+        {
+            KRATOS_ERROR << "Incorrect or missing ADAPTIVE_EXPONENT provided in process info for FrictionalViscoplasticTemperatureDependent2DLaw: "
+                         << rMaterialProperties[ADAPTIVE_EXPONENT] << std::endl;
+        }
+
+        if (rMaterialProperties[BULK_MODULUS] <= 0.0)
+        {
+            KRATOS_ERROR << "Incorrect or missing BULK_MODULUS provided in process info for FrictionalViscoplasticTemperatureDependent2DLaw: "
+                         << rMaterialProperties[BULK_MODULUS] << std::endl;
+        }
+
+        return 0;
+    }
+
+    double FrictionalViscoplasticTemperatureDependent2DLaw::GetEffectiveViscosity(ConstitutiveLaw::Parameters &rParameters) const
+    {
+        return rParameters.GetConstitutiveMatrix()(2, 2);
+    }
+
+    double FrictionalViscoplasticTemperatureDependent2DLaw::GetEffectiveDensity(ConstitutiveLaw::Parameters &rParameters) const
```

Comment:
honestly, you can unify these methods by adding an rVariable as an input in the method. Then this method should be named `GetEffectiveVariable(rParams, rVariable)`